src/riverroute/mosart_histfile.F90

```diff
@@ -479,6 +480,13 @@ subroutine htape_addfld (t, f, avgflag)
          call shr_sys_abort()
       end select
 
+      ! Override this field's avgflag if the namelist has set this tape to
+      ! - instantaneous
+      avgflag_temp = hist_avgflag_pertape(t)
+      if (avgflag_temp == 'I') then
+         tape(t)%hlist(n)%avgflag = avgflag_temp
+      end if
```

Comment:
Add these lines here and in https://github.com/ESCOMP/RTM/pull/39
as I did in the clm version of this code (https://github.com/ESCOMP/CTSM/pull/2838)?
```suggestion
      ! Override this tape's avgflag if nhtfrq == 1
      if (tape(t)%nhtfrq == 1) then  ! output is instantaneous
         hist_avgflag_pertape(t) = 'I'
      end if
      ! Override this field's avgflag if the namelist has set this tape to
      ! - instantaneous
      avgflag_temp = hist_avgflag_pertape(t)
      if (avgflag_temp == 'I') then
         tape(t)%hlist(n)%avgflag = avgflag_temp
      end if
```